kotlintest-runner/kotlintest-runner-jvm/src/main/kotlin/io/kotlintest/runner/jvm/TestCaseExecutor.kt

```diff
@@ -150,6 +150,10 @@ class TestCaseExecutor(private val listener: TestEngineListener,
       }
 
       supervisorJob.join()
+
+      // This cancel is necessary, or else the scheduler is reutilized by another test, and the shutdown will happen
+      // in the other test. This leads to tests "sharing" a timeout, and another test being timed out due to this task.
+      scheduledInterrupt.cancel(false)
```

Comment:
I'm confused by this. If the scheduled runnable still runs, all it does is shutdown the executor. The executor that will have already been shutdown, so what's the issue? There's no problem if it still runs right.

Reply:
I'm also pretty confused by this, as the Executor should be a different one every time. However, the issue seems to be on `TestEngineListener`, as it's shared between tests. `executor` will be `listenerExecutor` for single-threaded tests, and these will be interrupted as the executor shuts down

Reply:
Although I'm not completely sure if this is a `Threads` issue or a `Coroutines` issue of some sort, preventing the executor from shutting down if the test completes will fix the scenario that I created in `TestCaseExecutorTest`